src/__tests__/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,344 @@
+      ).container.firstChild,
+    )
+  }
+
+  const mutationsAndCallbacks = [
+    [makeMutationFn(), () => true],
+    [makeMutationFn(), () => getByTestId('the-element-we-are-looking-for')],
+    [
+      makeMutationFn(),
+      () => queryAllByTestId('the-element-we-are-looking-for'),
+    ],
+    [
+      () =>
+        container.removeChild(getByTestId('the-element-we-are-looking-for')),
+      () => getByTestId('the-element-we-are-looking-for'),
+    ],
+  ]
+
+  const callback = jest
+    .fn(() => {
+      throw new Error('No more calls are expected.')
+    })
+    .mockName('callback')
+    .mockImplementation(() => true)
+
+  const successHandler = jest.fn().mockName('successHandler')
+  const errorHandler = jest.fn().mockName('errorHandler')
+
+  const promise = waitForElementToBeRemoved(callback, {container}).then(
+    successHandler,
+    errorHandler,
+  )
+
+  // One synchronous `callback` call is expected.
+  expect(callback).toHaveBeenCalledTimes(1)
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  skipSomeTimeForMutationObserver()
+
+  // No more expected calls without DOM mutations.
+  expect(callback).toHaveBeenCalledTimes(1)
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  // Perform mutations one by one, waiting for each to trigger `MutationObserver`.
+  for (const [mutationImpl, callbackImpl] of mutationsAndCallbacks) {
+    callback.mockImplementation(callbackImpl)
+    mutationImpl()
+    skipSomeTimeForMutationObserver()
+  }
+
+  await promise
+
+  expect(callback).toHaveBeenCalledTimes(1 + mutationsAndCallbacks.length)
+  expect(successHandler).toHaveBeenCalledTimes(1)
+  expect(successHandler.mock.calls[0]).toMatchInlineSnapshot(`
+Array [
+  true,
+]
+`)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+})
+
+test('it waits characterData mutation', async () => {
+  const {container} = render(`<div>initial text</div>`)
+
+  const callback = jest
+    .fn(() => container.textContent === 'initial text')
+    .mockName('callback')
+  const successHandler = jest.fn().mockName('successHandler')
+  const errorHandler = jest.fn().mockName('errorHandler')
+
+  const promise = waitForElementToBeRemoved(callback, {container}).then(
+    successHandler,
+    errorHandler,
+  )
+
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+  expect(callback).toHaveBeenCalledTimes(1)
+
+  skipSomeTimeForMutationObserver()
+
+  expect(callback).toHaveBeenCalledTimes(1)
+  expect(successHandler).toHaveBeenCalledTimes(0)
+  expect(errorHandler).toHaveBeenCalledTimes(0)
+
+  container.querySelector('div').innerHTML = 'new text'
+  skipSomeTimeForMutationObserver()
+  await promise
+
+  expect(successHandler).toHaveBeenCalledTimes(1)
+  expect(successHandler.mock.calls[0]).toMatchInlineSnapshot(`
```

Comment:
Any reason this couldn't be simplified to:

```javascript
expect(successHandler).toHaveBeenCalledWith(true)
```

Or you could also replace this line and the one above with:

```javascript
expect(successHandler.mock.calls).toEqual([true])
```

I don't really care either way, but using snapshots for this doesn't make any sense to me personally.

Reply:
Yeah, make sense. I am a little  🆕 to `jest` that is the only reason.